consensus/block_validation.go

```diff
@@ -0,0 +1,69 @@
+package consensus
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/filecoin-project/go-filecoin/types"
+)
+
+type BlockValidator interface {
+	BlockSemanticValidator
+	BlockSyntaxValidator
+}
+
+type BlockSemanticValidator interface {
+	ValidateSemantic(ctx context.Context, child, parent *types.Block) error
+}
+
+type BlockSyntaxValidator interface {
+	ValidateSyntax(ctx context.Context, blk *types.Block) error
+}
+
+type BlockValidationClock interface {
+	Now() int64
+	BlockTime() time.Duration
+}
+
+type DefaultBlockValidationClock struct {
+	blkTime time.Duration
+}
+
+func NewDefaultBlockValidationClock(bt time.Duration) *DefaultBlockValidationClock {
+	return &DefaultBlockValidationClock{
+		blkTime: bt,
+	}
+}
+
+func (ebc *DefaultBlockValidationClock) Now() int64 {
+	return time.Now().Unix()
+}
+
+func (ebc *DefaultBlockValidationClock) BlockTime() time.Duration {
+	return ebc.blkTime
+}
+
+type DefaultBlockValidator struct {
+	clock BlockValidationClock
+}
+
+func NewDefaultBlockValidator(c BlockValidationClock) *DefaultBlockValidator {
+	return &DefaultBlockValidator{
+		clock: c,
+	}
+}
+
+func (dv *DefaultBlockValidator) ValidateSemantic(ctx context.Context, child, parent *types.Block) error {
+	// TODO validate timestamp
```

Comment:
File an issue for semantic validation and link to it. The spec is https://github.com/filecoin-project/specs/blob/master/validation.md (with clarification of syntax/semantics landing shortly).

Ditto for syntactic validation below.

Reply:
Done: #2886